src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java

```diff
@@ -0,0 +1,98 @@
+package com.greenfox.lvlup.model.entity;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import javax.persistence.*;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Data
+@EqualsAndHashCode(exclude = "holders")
+@Entity
+@Table(name = "badgelevels")
+public class BadgeLevel {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    private String name;
+    private int level;
+    private String description;
+
+    @ManyToOne
+    private Badge badge;
+
+    @ManyToMany
+    @JoinTable(name = "badgelevel_user",
+            joinColumns = @JoinColumn(name = "badgelevel_id", referencedColumnName = "id"),
+            inverseJoinColumns = @JoinColumn(name = "user_id", referencedColumnName = "id"))
+    private Set<User> holders;
+
+    public BadgeLevel() {
+    }
+
+    public BadgeLevel(String name, int level, String description, Badge badge, User... holders) {
+        this.name = name;
```

Comment:
As we've just discussed, the name field doesn't seem strictly necessary, retrieving and "forwarding" the name of the badge that belongs to any one badgelevel should be possible without storing this "duplicate" of it.

Reply:
I agree. Deleted name field in BadgeLevel:

public class BadgeLevel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private int level;
    private String description;